server/src/main/java/org/opensearch/snapshots/SnapshotsService.java

```diff
@@ -242,6 +243,11 @@ public SnapshotsService(
             clusterService.getClusterSettings()
                 .addSettingsUpdateConsumer(MAX_CONCURRENT_SNAPSHOT_OPERATIONS_SETTING, i -> maxConcurrentOperations = i);
         }
+
+        // Task will get retried from associated TransportClusterManagerNodeAction.
+        clusterService.registerThrottlingKey(ClusterManagerThrottlingKeys.UPDATE_SNAPSHOT_STATE_KEY, true);
+        clusterService.registerThrottlingKey(ClusterManagerThrottlingKeys.CREATE_SNAPSHOT_KEY, true);
+        clusterService.registerThrottlingKey(ClusterManagerThrottlingKeys.DELETE_SNAPSHOT_KEY, true);
```

Comment:
As discussed lets remove registration altogether since its no longer tied to the class and its more of a task type

Reply:
Replied the approach in : https://github.com/opensearch-project/OpenSearch/pull/4542#discussion_r986828995